### PR TITLE
Upload example data for AK Glacier demo; update example notebook accordingly; adjust P_rain/snow var updates

### PR DESCRIPTION
**Pull request by bolotinl**
**TL;DR**

- Uploaded the example data and config files for the glacier demo notebook to GH
- Updated the glacier demo notebook to show results using this exact example data/config files
- Changed the code that updates to `P_rain` and `P_snow` variables to allow compatibility with both gridded and non-gridded air temp data

**More Details:**

Previously, the example data and config files for running the [Glacier Demo notebook](https://github.com/NOAA-OWP/topoflow/blob/master/notebooks/TopoFlow_Glacier_AK_Demo.ipynb) was hosted on Google Drive for internal use. So that users can readily replicate this demo, **I uploaded the example data and configs to the** [examples folder](https://github.com/NOAA-OWP/topoflow/tree/master/topoflow/examples). Because the time varying grid of air temperature is too large to host on GitHub, **I updated the notebook to use a time series of air temperature that does not vary in space.** 

As briefly considered in [PR#15 to the peckhams/topoflow36](https://github.com/peckhams/topoflow36/pull/15) repo (but later reversed), the update of the P_rain and P_snow variables as-is causes issues if input air temperature data is not gridded, so **I also made a change to the line updating these variables in** `met_base.py`.  This solution does allocate the variable to new memory (not ideal), but for now may be the best solution to get TopoFlow running properly out-of-the-box for users following the demo notebook. Future development can help consider how we can reduce this potential inefficiency.

### CODE REVIEW

**Review by madMatchstick — APPROVED** [review]
@bolotinl - Thank you for the thorough and well organized PR.  I was able to run the AK_Demo notebook via the "quick start" option.  Just a few minor suggestions...

1.  `from osgeo` was needed to `import gdal, osr` in `topoflow/utils/regrid.py`
2.  Recommend adding some disclosure in the Run Topoflow block mentioning runtime.  It doesn't need to be overly descriptive but something along the lines of "Be aware that this simulation will run through time 525,600.0 [min] and will likely take a few hours to complete."
3. This is probably obvious to those well-versed in world of `jupyter notebooks`  but the user will need to first activate `tf36` conda environment (or similar) to run the notebook :) Probably wouldn't hurt to add that somewhere at the top.

-thanks!

10.30.2024 edit - Note I was also able to run all pre-processing steps outlined in Option 2 in the demo.ipynb

**Comment by bolotinl** [comment]
@madMatchstick thanks for the comments and feedback Jess! Per your suggestions: 
* I added `from osgeo` to the gdal import in regrid.py
* I added a disclaimer to the notebook telling people how long it takes to run and how they could shorten the run if they so please
* At the top of the notebook, I added a link to Scott's instructions for installing topoflow, setting up the environment and Jupyter notebooks, and added a quickstart to all of these tasks to the notebook itself.
